packages/@aws-cdk/aws-ec2/lib/cloudformation-init.ts

```diff
@@ -0,0 +1,225 @@
+import cdk = require('@aws-cdk/cdk');
+
+/**
+ * Describe startup configuration for EC2 instances
+ */
+export class CloudFormationInit extends cdk.Construct {
+  /**
+   * Logical ID of the Init resource
+   */
+  public readonly initResourceId: string;
+
+  private readonly configs: {[name: string]: InitConfig} = {};
+  private readonly configSets: {[name: string]: string[]} = {};
+
+  constructor(parent: cdk.Construct, id: string) {
+    super(parent, id);
+
+    const resource = new cdk.Resource(this, 'Resource', {
+      // I know this thing says it lives in the AWS::CloudFormation namespace,
+      // but it really doesn't make sense there.
+      type: 'AWS::CloudFormation::Init',
```

Comment:
Something is weird. I believe `AWS::CloudFormation::Init` is a metadata key, not a resource type:
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-init.html

Reply:
Doh, you're right! Completely misread that.